### ISSUE TITLE
Crash when you switching network from Mainnet to Ropsten (reproduced only once)

### ISSUE THREAD

**Issue by churik** [status-im/status-react#4913]
### Description

*Type*: Bug

*Summary*: reproduced once on IOS when switched from Mainnet to Ropsten (have active transactions, which are receiving confirmations on Mainnet) 

#### Expected behavior
no crash

#### Actual behavior
crash

### Reproduction

- оpen Status
- send SNT to another user in 1-1 chat
- open Wallet->Transaction History (transaction is not finished yet, received confirmations)
- open profile
- tap on Advanced
- change network to Ropsten with upstream RPC
- log in

### Additional Information
* Status version: 
* Operating System: iOS

#### Logs
* TF session(from 24:22): https://app.testfairy.com/projects/4803590-status/builds/8503790/sessions/4394603372/?accessToken=9bP0E7-P34sNNiY-etWBt7L6yCw

Crash log:
```
0   StatusIm                            0x0000000103526d68 0x102b9c000 + 10005864
1   StatusIm                            0x0000000103526968 0x102b9c000 + 10004840
2   StatusIm                            0x00000001037683d0 0x102b9c000 + 12370896
3   libdispatch.dylib                   0x0000000184020aa0 0x18401f000 + 6816
4   libdispatch.dylib                   0x0000000184020a60 0x18401f000 + 6752
5   libdispatch.dylib                   0x0000000184027b84 0x18401f000 + 35716
6   libdispatch.dylib                   0x000000018402dcac 0x18401f000 + 60588
7   libdispatch.dylib                   0x000000018402d9fc 0x18401f000 + 59900
8   libsystem_pthread.dylib             0x0000000184353fac 0x184353000 + 4012
9   libsystem_pthread.dylib             0x0000000184353b08 0x184353000 + 2824
```

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/8503790/sessions/4394603372/?accessToken=9bP0E7-P34sNNiY-etWBt7L6yCw">TestFairy: Status 0.9.21 (6106) @2018</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@churik isn't it a duplicate of https://github.com/status-im/status-react/issues/3981?

**Comment by churik** [comment]
Added info to  #3981
